src/Arcus.Messaging.Tests.Integration/MessagePump/ServiceBusMessagePumpTests.cs

```diff
@@ -763,6 +766,85 @@ public async Task ServiceBusTopicMessagePumpWithServiceBusAbandonAfterContextPre
             await TestServiceBusTopicMessageHandlingForW3CAsync(options);
         }
 
+         [Fact]
+        public async Task ServiceBusTopicMessagePump_PauseViaCircuitBreaker_RestartsAgainWithOneMessage()
+        {
+            // Arrange
+            var options = new WorkerOptions();
+            ServiceBusMessage[] messages = GenerateShipmentMessages(3);
+            TimeSpan recoveryTime = TimeSpan.FromSeconds(5);
+            TimeSpan messageInterval = TimeSpan.FromSeconds(1);
+
+            options.AddXunitTestLogging(_outputWriter)
+                   .AddServiceBusTopicMessagePump(
+                       subscriptionName: "circuit-breaker-" + Guid.NewGuid(),
+                       _ => _config.GetServiceBusTopicConnectionString())
+                   .WithServiceBusMessageHandler<CircuitBreakerAzureServiceBusMessageHandler, Shipment>(
+                        implementationFactory: provider => new CircuitBreakerAzureServiceBusMessageHandler(
+                            targetMessageIds: messages.Select(m => m.MessageId).ToArray(),
+                            configureOptions: opt =>
+                            {
+                                opt.MessageRecoveryPeriod = recoveryTime;
+                                opt.MessageIntervalDuringRecovery = messageInterval;
+                            },
+                            provider.GetRequiredService<IMessagePumpCircuitBreaker>()));
+
+            var producer = TestServiceBusMessageProducer.CreateFor(_config, ServiceBusEntityType.Topic);
+            await using var worker = await Worker.StartNewAsync(options);
+
+            // Act
+            await producer.ProduceAsync(messages);
+
+            // Assert
+            var handler = GetMessageHandler<CircuitBreakerAzureServiceBusMessageHandler>(worker);
+            AssertX.RetryAssertUntil(() =>
+            {
+                DateTimeOffset[] arrivals = handler.GetMessageArrivals();
+                Assert.Equal(messages.Length, arrivals.Length);
+
+                TimeSpan faultMargin = TimeSpan.FromSeconds(1);
+                Assert.Collection(arrivals.SkipLast(1).Zip(arrivals.Skip(1)),
+                    dates => AssertDateDiff(dates.First, dates.Second, recoveryTime, recoveryTime.Add(faultMargin)),
+                    dates => AssertDateDiff(dates.First, dates.Second, messageInterval, messageInterval.Add(faultMargin)));
+
+            }, timeout: TimeSpan.FromMinutes(2), _logger);
+        }
```

Comment:
Test checks that we both use the message recovery time and message interval time correctly. The message handler used here will for the first received message call the message circuit breaker (which will start the message recovery period), and on the first received message throw an exception (which will start the message interval period). The third message will be handled correctly.

In the assertion, we check that the time between these three messages is the expected time configured in the message circuit breaker.